### PR TITLE
Update deployment titles 

### PR DESCRIPTION
**Pull request by Deflaimun**
Resolves https://github.com/redpanda-data/documentation-private/issues/2302

Preview:
https://deploy-preview-347--redpanda-docs-preview.netlify.app/current/deploy/deployment-option/self-hosted/manual/production/production-deployment-automation/

https://deploy-preview-347--redpanda-docs-preview.netlify.app/current/deploy/deployment-option/self-hosted/manual/production/production-deployment/

### CODE REVIEW

File: modules/deploy/pages/deployment-option/self-hosted/manual/production/production-deployment.adoc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-= Deploy for Production
+= Manual Deployment for Production
```
Comment (Deflaimun) [inline modules/deploy/pages/deployment-option/self-hosted/manual/production/production-deployment.adoc]:
```suggestion
= Manual Deploy for Production
```

**Comment by Deflaimun** [comment]
note, we don´t usually change the file names (that directly affect the URL) unless we have a very strong reason to avoid redirects

**Comment by micheleRP** [comment]
@Feediver1, @hcoyote: This feels weird to me. Could we do:
- Deploy for Production: Automated
- Deploy for Production: Manual

**Comment by hcoyote** [comment]
> @Feediver1, @hcoyote: This feels weird to me. Could we do:
> 
> * Deploy for Production: Automated
> * Deploy for Production: Manual

I'm not wedded to the specific wording, just the outcome of disambiguating and clarifying that one is for automated deployments and one is for doing it manually.

File: modules/deploy/pages/deployment-option/self-hosted/manual/production/production-deployment-automation.adoc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-= Automated Deployment
+= Automated Deploy for Production
```
Comment (Deflaimun) [inline modules/deploy/pages/deployment-option/self-hosted/manual/production/production-deployment-automation.adoc]:
```suggestion
= Deploy for Production: Automated
```

File: modules/deploy/pages/deployment-option/self-hosted/manual/production/production-deployment.adoc [hunk 3]
```diff
@@ -1,4 +1,4 @@
-= Deploy for Production
+= Manual Deploy for Production
```
Comment (Deflaimun) [inline modules/deploy/pages/deployment-option/self-hosted/manual/production/production-deployment.adoc]:
```suggestion
= Deploy for Production: Manual
```

**Review by micheleRP — APPROVED** [review]
lgtm

**Pull request merged**